src/Tasks/ListOperators/RemoveDuplicates.cs

```diff
@@ -15,47 +14,49 @@ namespace Microsoft.Build.Tasks
     /// </summary>
     public class RemoveDuplicates : TaskExtension
     {
-        private ITaskItem[] _inputs = Array.Empty<TaskItem>();
-        private ITaskItem[] _filtered = null;
-
         /// <summary>
         /// The left-hand set of items to be RemoveDuplicatesed from.
         /// </summary>
-        public ITaskItem[] Inputs
-        {
-            get { return _inputs; }
-            set { _inputs = value; }
-        }
+        public ITaskItem[] Inputs { get; set; } = Array.Empty<TaskItem>();
 
         /// <summary>
         /// List of unique items.
         /// </summary>
         [Output]
-        public ITaskItem[] Filtered
-        {
-            get { return _filtered; }
-            set { _filtered = value; }
-        }
+        public ITaskItem[] Filtered { get; set; } = null;
+
+        /// <summary>
+        /// True if any duplicate items were found. False otherwise.
+        /// </summary>
+        [Output]
+        public bool HadAnyDuplicates { get; set; } = false;
 
         /// <summary>
         /// Execute the task.
         /// </summary>
         /// <returns></returns>
         public override bool Execute()
         {
-            Hashtable alreadySeen = new Hashtable(Inputs.Length, StringComparer.OrdinalIgnoreCase);
-            ArrayList filteredList = new ArrayList();
+            if (Inputs == null || Inputs.Length == 0)
```

Comment:
The empty check here makes this implementation faster than the original one on empty inputs. Is such complexity worth it?

* New: without empty check (always allocate `HashSet`, but never enter the loop)
* NewEmpty: with empty check
* Old: the `Hashtable` implementation

```
       Empty |          Clr | LegacyJit |      New |  9.904 ms | 0.1030 ms | 0.0861 ms |  9.890 ms |
             |              |           |          |           |           |           |           |
       Empty |          Clr | LegacyJit | NewEmpty |  9.746 ms | 0.0083 ms | 0.0078 ms |  9.747 ms |
             |              |           |          |           |           |           |           |
       Empty |          Clr | LegacyJit |      Old |  9.861 ms | 0.0893 ms | 0.0836 ms |  9.830 ms |
             |              |           |          |           |           |           |           |
       Empty | LegacyJitX86 | LegacyJit |      New |  9.842 ms | 0.0278 ms | 0.0232 ms |  9.837 ms |
             |              |           |          |           |           |           |           |
       Empty | LegacyJitX86 | LegacyJit | NewEmpty |  9.656 ms | 0.0138 ms | 0.0116 ms |  9.655 ms |
             |              |           |          |           |           |           |           |
       Empty | LegacyJitX86 | LegacyJit |      Old |  9.709 ms | 0.0916 ms | 0.0765 ms |  9.681 ms |
             |              |           |          |           |           |           |           |
       Empty |    RyuJitX86 |    RyuJit |      New |  9.720 ms | 0.0215 ms | 0.0201 ms |  9.716 ms |
             |              |           |          |           |           |           |           |
       Empty |    RyuJitX86 |    RyuJit | NewEmpty |  9.675 ms | 0.0309 ms | 0.0274 ms |  9.666 ms |
             |              |           |          |           |           |           |           |
       Empty |    RyuJitX86 |    RyuJit |      Old |  9.823 ms | 0.0174 ms | 0.0163 ms |  9.819 ms |
```